### DISCUSSION TITLE
Disable asciify and honor `replace` rules in metadata fields?

### DISCUSSION THREAD

**Discussion by treyturner** [gtronset/beets-filetote#161] [Ideas]
Hi; great plugin, thanks!

I have some rules for the `replace` plugin that serve a few purposes:
- convert symbols commonly used in artist, album, and track names but which are illegal in most filesystems to filesystem-friendly unicode equivalents
- convert unnecessarily unicode symbols (dashes, apostrophes and quotes) into ascii equivalents
- handle a few edge cases

It seems filetote filenames are run through `%asciify`, as unicode characters end up lossy converted to `_`. Just curious if it would be reasonable to allow this behavior to be optionally disabled, and for the replace rules to be applied if they aren't already. Thanks for your consideration!

```
path_sep_replace: "\u29F8" #  force / to become ⧸ (special case for linux host)

filetote:
  exclude: proof.jpg proof.jpeg proof.png
  extensions: .cue .jpg .jpeg .log .nfo .pdf .png .srr
  filenames: file_id.diz
  paths:
    ext:.cue: $albumpath/$subpath$albumartist - $album
    ext:.log: $albumpath/$subpath$albumartist - $album
    ext:.nfo: $albumpath/$subpath$albumartist - $album
    ext:.pdf: $albumpath/$subpath$albumartist - $album
    ext:.srr: $albumpath/$subpath$albumartist - $album    
    ext:.jpeg: $albumpath/$subpath$old_filename
    ext:.jpg: $albumpath/$subpath$old_filename
    ext:.png: $albumpath/$subpath$old_filename    

replace:  
  '/': "\u29F8"  # / becomes ⧸
  '\\': "\u29F9"  # \ becomes ⧹
  '\?': "\uFF1F"  # ? becomes ？
  ':': "\u2236"  # : becomes ∶
  '\.\.\.': "\u2026"  # ... becomes …
  '^\.': '' # remove leading period to prevent creating hidden files
  '<': "\u2039"  # < becomes ‹
  '>': "\u203A"  # > becomes ›
  '\*': "\u2731"  # * becomes ✱
  '\|': "\uFF5C"  # | becomes ｜
  '[\x00-\x1F]': ''  # remove control/unprintable characters
  '\.$': ''  # remove trailing dot before extension
  '\s+$': '' # remove trailing whitespace 
  '[\u2010-\u2015]': '-'  # convert unicode dashes to ascii
  '[\u2018-\u201B]': '''' # convert unicode apostrophes to ascii
  '[\u201C-\u201F]': '"' # convert unicode quotes to ascii
```

**Comment by gtronset** [comment]
@treyturner great catch, and thank you for the detailed information here. This was resolved in https://github.com/gtronset/beets-filetote/pull/177 and made available in Filetote [`v1.0.1`](https://github.com/gtronset/beets-filetote/releases/tag/v1.0.1). 

Please note that these fixes will only apply to the newest version of Filetote, which only supports Beets v2+.
